src/auth/user-record.ts

```diff
@@ -57,19 +58,226 @@ export interface CreateRequest extends UpdateRequest {
+    return {
+      uid: this.uid,
+      displayName: this.displayName,
+      factorId: this.factorId,
+      enrollmentTime: this.enrollmentTime,
+    };
+  }
+
+  /**
+   * Returns the factor ID based on the response provided.
+   *
+   * @param response The server side response.
+   * @return The multi-factor ID associated with the provided response. If the response is
+   *     not associated with any known multi-factor ID, null is returned.
+   */
+  protected abstract getFactorId(response: MfaInfo): MultiFactorId | null;
+
+  /**
+   * Initializes the MultiFactorInfo object using the provided server response.
+   *
+   * @param response The server side response.
+   */
+  private initFromServerResponse(response: MfaInfo, factorId: MultiFactorId | null) {
+    if (factorId && response && response.mfaEnrollmentId) {
+      utils.addReadonlyGetter(this, 'uid', response.mfaEnrollmentId);
+      utils.addReadonlyGetter(this, 'factorId', factorId);
+      utils.addReadonlyGetter(this, 'displayName', response.displayName || null);
+      // Encoded using [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format.
+      // For example, "2017-01-15T01:30:15.01Z".
+      // This can be parsed directly via Date constructor.
+      // This can be computed using Data.prototype.toISOString.
+      if (response.enrolledAt) {
+        utils.addReadonlyGetter(
+            this, 'enrollmentTime', new Date(response.enrolledAt).toUTCString());
+      } else {
+        utils.addReadonlyGetter(this, 'enrollmentTime', null);
+      }
+    } else {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INTERNAL_ERROR,
+        'INTERNAL ASSERT FAILED: Invalid multi-factor info response');
+    }
+  }
+}
+
+/** Class representing a phone MultiFactorInfo object. */
+export class PhoneMultiFactorInfo extends MultiFactorInfo {
+  public readonly phoneNumber: string;
+
+  /**
+   * Initializes the PhoneMultiFactorInfo object using the server side response.
+   *
+   * @param response The server side response.
+   * @constructor
+   */
+  constructor(response: MfaInfo) {
+    super(response);
+    utils.addReadonlyGetter(this, 'phoneNumber', response.phoneInfo);
+  }
+
+  /** @return The plain object representation. */
+  public toJSON(): any {
+    return Object.assign(
+        super.toJSON(),
+        {
+          phoneNumber: this.phoneNumber,
+        });
+  }
+
+  /**
+   * Returns the factor ID based on the response provided.
+   *
+   * @param response The server side response.
+   * @return The multi-factor ID associated with the provided response. If the response is
+   *     not associated with any known multi-factor ID, null is returned.
+   */
+  protected getFactorId(response: MfaInfo): MultiFactorId | null {
+    return !!response.phoneInfo ? MultiFactorId.Phone : null;
+  }
+}
+
+/** Class representing multi-factor related properties of a user. */
+export class MultiFactor {
+  public readonly enrolledFactors: ReadonlyArray<MultiFactorInfo>;
```

Comment:
This will be a lot clear if we rename `MultiFactorInfo` to `AuthFactor`.

"`MultiFactor` has a collection of `AuthFactor`s" has a nice ring to it.

Reply:
Same here. This can't be changed at this point. It is the same name used in all the client SDKs. The Admin Node.js implementation should align with the JS client implementation.